data_transfer/devices/dreem.py

```diff
@@ -108,33 +112,34 @@ def download_metadata(self) -> None:
                     )
                 # Edge-case: device not logged with patient in UCAM
                 else:
-                    print(f"Metadata cannot be determined for:\n    {recording}")
+                    log.debug(
+                        f"Device does not exists in UCAM for {patient_id}:\n {recording}"
+                    )
+                    log.debug("Attempting to use inventory.")
                     continue
             else:
                 unknown += 1
-                print(f"Metadata cannot be determined for:\n    {recording}")
+                log.debug(f"Patient ID ({patient_id}) is not in UCAM:\n    {recording}")
```

Comment:
Could you please implement the ID formatter and validator - as [in the ByteFlies branch?](https://github.com/ideafast/middleware-services/blob/3e65fca9a451f1fb4599b795e823709159683a77/data_transfer/devices/byteflies.py#L113-L115). This also shows that we keep the `-` throughout the pipeline and only remove it when translating to uploads to the DMP.